lib/tools/system-calls.js

```diff
@@ -779,31 +795,19 @@ systemCallMethods.waitForDevice = async function (appDeviceReadyTimeout = 30) {
  */
 systemCallMethods.reboot = async function (retries = DEFAULT_ADB_REBOOT_RETRIES) {
   try {
-    try {
-      await this.shell(['stop']);
-    } catch (err) {
-      if (err.message.indexOf('must be root') === -1) {
-        throw err;
-      }
-      // this device needs adb to be running as root to stop.
-      // so try to restart the daemon
-      log.debug('Device requires adb to be running as root in order to reboot. Restarting daemon');
-      await this.root();
-      await this.shell(['stop']);
-    }
-    await B.delay(2000); // let the emu finish stopping;
+    await this.privilegedShell(['stop']);
+    await B.delay(EMU_STOP_TIMEOUT); // let the emu finish stopping;
```

Comment:
so, cannot we have a timer here?

Reply:
@mykola-mokhnach So you mean to use `setTimeout` instead of `delay`?

Reply:
`B.delay` is a timer, isn't it?

Reply:
Im confused about this too 😄

Reply:
I mean something that check for the expected results and waits if the result is not there or exists immediately, like the `retry` or `waitForCondition` functions, instead of the hardcoded delay

Reply:
@mykola-mokhnach the reboot is going to wait, right now I don't know what condition we could be using for checking stop finished, but I feel is out of the scope of this PR.